fix(curriculum): Recipe Tracker improvements (#57694)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-recipe-tracker/66fbcf750a62784cf11f5630.md

Use the `push()` method to push all the recipe objects into the `recipes` array. Make sure to push `recipe1`, `recipe2`, and `recipe3` in that order.

> Also delete `recipe1Name`, `recipe2Name`, `recipe1CookingTime`, and `recipe2CookingTime` variables and their logging to console.

Reviewer: ```suggestion
Also delete the `recipe1Name`, `recipe2Name`, `recipe1CookingTime`, and `recipe2CookingTime` variables, and the `console.log` statements which log those variables.
```

# --hints--

> You should remove `recipe1Name` variable.

Reviewer: ```suggestion
You should remove the `recipe1Name` variable.
```

```js

> assert.notMatch(__helpers.removeJSComments(code), /const\s+recipe1Name\s*=\s*recipe1\s*\.\s*name\s*;?/);

Reviewer: ```suggestion
assert.isUndefined(recipe1name);
```

```

> You should remove `console.log(recipe1Name);`.

Reviewer: ```suggestion
You should remove the `console.log(recipe1Name);` statement.
```

```js
assert.notMatch(__helpers.removeJSComments(code), /console\s*\.\s*log\s*\(\s*recipe1Name\s*\)\s*;?/);
```

> You should remove `recipe2Name` variable.

Reviewer: ```suggestion
You should remove the `recipe2Name` variable.
```

```js

> assert.notMatch(__helpers.removeJSComments(code), /const\s+recipe2Name\s*=\s*recipe2\s*\.\s*name\s*;?/);

Reviewer: ```suggestion
assert.isUndefined(recipe2name);
```

I think these will work, but it's possible they will throw a reference error. I can't remember.

Author: Unfortunately it will cause `ReferenceError`. We could do it with `try/catch` instead, ie:

```js
try {
  recipe1Name;
  assert.fail('Variable is defined/declared');
} catch (e) {
  assert.include(e?.message, 'recipe1Name is not defined');
}
```

What do you think?

Reviewer: That works!

```

> You should remove `console.log(recipe2Name);`.

Reviewer: ```suggestion
You should remove the `console.log(recipe2Name);` statement.
```

```js
assert.notMatch(__helpers.removeJSComments(code), /console\s*\.\s*log\s*\(\s*recipe2Name\s*\)\s*;?/);
```

> You should remove `recipe1CookingTime` variable.

Reviewer: ```suggestion
You should remove the `recipe1CookingTime` variable.
```

```js

> assert.notMatch(__helpers.removeJSComments(code), /const\s+recipe1CookingTime\s*=\s*recipe1CookingTime\s*\.\s*name\s*;?/);

Reviewer: ```suggestion
assert.isUndefined(recipe2name);
```

```

> You should remove `console.log(recipe1CookingTime);`.

Reviewer: ```suggestion
You should remove the `console.log(recipe1CookingTime);` statement.
```

```js
assert.notMatch(__helpers.removeJSComments(code), /console\s*\.\s*log\s*\(\s*recipe1CookingTime\s*\)\s*;?/);
```

> You should remove `recipe2CookingTime` variable.

Reviewer: ```suggestion
You should remove the `recipe2CookingTime` variable.
```

```js
assert.notMatch(__helpers.removeJSComments(code), /const\s+recipe2CookingTime\s*=\s*recipe2CookingTime\s*\.\s*name\s*;?/);
```

> You should remove `console.log(recipe2CookingTime);`.

Reviewer: ```suggestion
You should remove the `console.log(recipe2CookingTime);` statement.
```
